renku/infrastructure/storage/base.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2022- Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base storage handler."""
+
+import os
+import subprocess
+from typing import Any, List
+
+from renku.core import errors
+from renku.core.interface.storage import IStorage
+
+
+class BaseStorage(IStorage):
+    """Base external storage handler class."""
```

Comment:
This currently has a lot of rclone-specific things. Should it then be called `RcloneBaseStorage` or something like that. Or if this is supposed to be more general then the rclone-specific stuff should then move someplace else?